### DISCUSSION TITLE
No new devices in Domoticz

### DISCUSSION THREAD

**Discussion by Marmachine** [zwave-js/zwave-js-ui#3502] [General]
I am succesfull in moving to Zwave-JS-UI so far...
Until today i've had my Aeotec Z-stick 5G integrated with Domoticz, now i see my nodes in MQTT Explorer, so far so good.
Now i am expecting new devices to popup in Domoticz, but they don't.
I have checked configuration, but i don't see what I might have missed.
Ohh... and yes i have checked "accept new hardware" setting

**Comment by robertsLando** [comment]
Did you enabled mqtt discovery?

**Reply by Marmachine** [reply]
Where do i do that? In Domoticz i don't see that option... do i first need an update for that? (i'm on 2023.1)

If you mean the setting in Z-Wave JS UI settings, under Home Assistant, that's on yes. (however, i am on Domoticz)

**Comment by akamming** [comment]
and did you enter the actual IP adress (so no localhost or 127.0.0.1) as the host adress for the MQTT server?

**Reply by Marmachine** [reply]
Yes, i will try to add some more info later today.

**Comment by FriedlNetDZ** [comment]
Please give me some more information about your settings in Z-Wave JS UI and in Domoticz. I had exactly the same error pattern.

**Reply by Marmachine** [reply]
I think (not sure!) the problem is with the Zwave serial port (the configuration for the Zstick Gen5)...
Since I have persistent USB configuration, this takes care of recognizing the Zstick, no matter in which port it is.
For Domoticz this allows to assign as serial port /dev/ttyUSB-ZStick-5G to make it work.
However, i figured that i should leave this as is and just follow the instructions;

The docker-compose.yml configuration file example shows the serial port configuration:
devices:
      - '/dev/serial/by-id/usb-Silicon_Labs_CP2102_USB_to_UART_Bridge_Controller_0001-if00-port0:/dev/zwave'
It is important to note that **you need to replace the serial port part** with the correct serial device you found before.
(keep the /dev/serial/by-id/ in front) So this is what i did;

What i had found earlier for my Zstick was this: usb-0658_0200-if00 (so, this is what I had put after /dev/serial/by-id/ according to the instructions, resulting in: /dev/serial/by-id/usb-0658_0200-if00

Looking at the example yml file again, i wonder if this should be followed by _-port0:/dev/zwave_ like the example, this to allow communication to (interview) the Zstick? Changing the settings now doesn't do anything though.

Serial port in Zwave-JS-UI /dev/serial/by-id/usb-0658_0200-if00 (or /dev/ttyUSB-ZStick-5G) both result in the following error:
Driver: Failed to open the serial port: Error: No such device or address, cannot open /dev/serial/by-id/usb-0658_0200-if00 (ZW0100) 
 
Also in debug I see:
2024-01-05 17:29:17.276 DEBUG MQTT: Publishing to zwave/_EVENTS/ZWAVE_GATEWAY-zwave-js-ui/driver/driver_error: {
  data: [
    ZWaveError: Failed to open the serial port: Error: No such device or address, cannot open /dev/serial/by-id/usb-0658_0200-if00 (ZW0100)
        at Driver.openSerialport (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1286:9)
        at Immediate.<anonymous> (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1121:5) {
      [stack]: 'ZWaveError: Failed to open the serial port: Error: No such device or address, cannot open /dev/serial/by-id/usb-0658_0200-if00 (ZW0100)\n' +
        '    at Driver.openSerialport (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1286:9)\n' +
        '    at Immediate.<anonymous> (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1121:5)',
      message: 'Failed to open the serial port: Error: No such device or address, cannot open /dev/serial/by-id/usb-0658_0200-if00 (ZW0100)',
      code: 100,
      context: undefined,
      transactionSource: undefined,
      name: 'ZWaveError'
    },
    [length]: 1
  ]
} with options { qos: 1, retain: false } 
 
I don't know where to go from here, so maybe someone can give me some direction with this?

**Reply by kpine** [reply]
Your USB path config in ZUI should be `/dev/zwave` if you are using the quoted example.

e.g. the configuration:

```yaml
devices:
- '/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave'
```

Means the path inside the container is `/dev/zwave`.

**Reply by Marmachine** [reply]
OK, so i update the line in docker-compose.yml file, devices to:
      - '/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave'

Then it should work? Should i stop/start/restart anything then?
I have my device deactivated in Domoticz, all those nodes and sensors are now not showing in Domoticz, only under settings > devices i can find them. Already noticed that, when i activate the Zstick under hardware, they all come back though... i need that to replace them with new devices (once they're added).

**Reply by kpine** [reply]
It doesn't matter what name you give the path inside the container, it just needs to match in the ZUI configuration.

```yaml
devices:
- '/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave'
```
then configure ZUI with `/dev/zwave`.

```yaml
devices:
- '/dev/serial/by-id/usb-0658_0200-if00'
```
then configure ZUI with `/dev/serial/by-id/usb-0658_0200-if00`.

The first one is just simpler to configure inside ZUI and more portable (if you restore NVM backup to another stick, for example).

Not sure why you thought your USB configuration was at fault initially though. Your custom udev rule works the same as the `/dev/serial/by-id` rule, it's just not a default configuration. Any problems with device path configuration would be reflected in ZUI.

**Reply by Marmachine** [reply]
Sorry... let's not talk different options while i'm not understanding this (its still confusing me).
I just configured my .yml file with this
`devices:
      - '/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave'`
      
Now in ZUI in Settings > Zwave > I need to enter the Serial Port: /dev/serial/by-id/usb-0658_0200-if00

Then it should work? No reboot or anything?

**Reply by kpine** [reply]
No, that would be backwards.

![image](https://github.com/zwave-js/zwave-js-ui/assets/791794/c1ebf883-1741-426d-97de-848cb3114229)

`/dev/zwave` is the path inside the container, and you always configure ZUI's serial port path using the inside container path.

```yaml
devices:
- '<path on host>:<path inside container>'
```

then configure ZUI with `<path inside container>`.

**Reply by Marmachine** [reply]
I'm confused, is this what you mean:

![image](https://github.com/zwave-js/zwave-js-ui/assets/25507805/cbf8c1b8-dd50-481c-b8a8-b76eb5d92d8d)

**Reply by kpine** [reply]
Yes, that's exactly what I wrote. :wink:

**Reply by Marmachine** [reply]
Still an error: Failed to open the serial port...
Along the way i've lost all my nodes in ZUI as well... it was a wonder in the first place, don't know what i did to get them.
Now what... does it need time, reboot or what can i do to get this thing running?

**Reply by kpine** [reply]
Did you re-create the container after changing the yaml to use device `/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave`?

**Reply by Marmachine** [reply]
Here's what i have in OpenZwave Control Panel
![image](https://github.com/zwave-js/zwave-js-ui/assets/25507805/8c939755-361f-4da1-8bc4-a89bdf206044)

**Reply by kpine** [reply]
Why are you using OpenZwave Control Panel? You can't run both software at the same time.

**Reply by Marmachine** [reply]
Oh, okay... i need to run **docker compose up -d** again? First docker compose down?

**Reply by kpine** [reply]
Seems you got it already, but yes, run `docker compose up -d` anytime you change the compose file.

**Comment by Marmachine** [comment]
Well... okay, devices are back, most remain undefined but i see their messages in MQTT Explorer
Now how would i get them added as new devices into Domoticz?

![image](https://github.com/zwave-js/zwave-js-ui/assets/25507805/4341e5ea-ddd0-478e-a0b6-2ea5b090444d)

**Reply by kpine** [reply]
> okay, devices are back, most remain undefined

I'd suggest posting your entire docker compose file. It looks like you've mis-configured the store volume as well.

Although, they look like battery devices, so perhaps not. In that case, you will need to manually wake them up to perform interviews for the first time.

**Reply by Marmachine** [reply]
I have three sitting right at my desk here... actually one (NodeID 003) is powered by USB (its an Aeotec Multisensor), two others (are some chineese brand) and battery powered. They blink and send a message on motion through MQTT, so their basic functionality seems okay. I was expecting the powered Aeotec to announce itself, but generally i'll look into waking up all of them. 

So i guess the old configuration (like i showed from OpenZwave Control Panel) isn't somehow re-used then i guess? 

Here's my docker compose file:

```
version: "3.7"
services:
  zwave-js-ui:
    container_name: zwave-js-ui
    image: zwavejs/zwave-js-ui:latest
    restart: unless-stopped
    tty: true
    stop_signal: SIGINT
    environment:
      - SESSION_SECRET=mySecretSession
      - TZ=Europe/Amsterdam
    devices:
      - '/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave'
    volumes:
      - ./store:/usr/src/app/store
    ports:
      - "8091:8091" # port for web interface
      - "3000:3000" # port for Z-Wave JS websocket server
```

**Reply by kpine** [reply]
Compose file looks good. You don't need port 3000 though if you are only using MQTT, that is for the Home Assistant Z-Wave integration.

You must have included node 3 on battery power? Because it is acting like a battery device. You can't include on battery power then add USB power later, you won't get mains-powered functionality.

The battery devices might not function properly until they are interviewed.

**Reply by Marmachine** [reply]
Okay, great! All there now:
![image](https://github.com/zwave-js/zwave-js-ui/assets/25507805/f49fa52d-ff88-4024-8b92-4ca596a696f7)

So now how to get these as new devices in Domoticz? ("accept new hardware" is ON)

**Reply by kpine** [reply]
> So now how to get these as new devices in Domoticz? ("accept new hardware" is ON)

Domoticz has it's own documentation, did you check there? https://www.domoticz.com/wiki/Zwave-JS-UI#Installing_ZWaveJS-UI

Probably better to ask for additional help in the Domoticz support forums if you still have issues with integrating it.

**Comment by FriedlNetDZ** [comment]
Is it working now or do you have problems with domoticz?

**Reply by Marmachine** [reply]
The previous was very helpfull to understand and get my nodes in MQTT.
I haven't found out how to get my devices in Domoticz... no new devices have been added, old devices have disapeared.
Already had hardware configured in Domoticz: MQTT Client Gateway > MQTT Auto Discovery Client Gateway with LAN interface 
So something doesn't seem to be working right ...?

**Comment by akamming** [comment]
best way to troubleshoot is  the following order:
A) try to get the devices visible in ZWAVE JS UI --> most common cause why not visible: Something wrong with configured serial port for ZWAVE gateway
B) download MQTT explorer (https://mqtt-explorer.com/) en connect to your mqtt server and check if you can see the discovery topic. If you cannot connect: Troubleshoot your mqtt server until you can connect to your mqtt server. If you can connect, but do no see the devices: Check you mqtt settings zwave js ui until you see them in MQTT explorer
C) Check if devices are visibile in domoticz. If you cannot see them:
- check if you domoticz client connects to mqtt server  (if it does not, check your mqtt settin
- if and if does: Check if autodiscovery is enabled in both domoticz and zwave js ui and if the discovery topic is correctly entered in domoticz setup (should be the same as the discovery topic in zwave js ui)

good luck!

**Reply by Marmachine** [reply]
Yeah, all is check, but no devices.
Through Node-red i can send MQTT messages to toggle a (virtual) device on and off in Domoticz.
Node-red is triggered on motion messages from nodes that publish through ZWAVE JS UI to MQTT
So i see the Nodes in MQTT (also confirmed through MQTT Explorer) and Domoticz is connected to the broker since i can toggle a switch through MQTT...
I have found a wiki page, https://www.domoticz.com/wiki/MQTT#Add_hardware_.22MQTT_Auto_Discovery_Client_Gateway.22 but it didn't bring any news until my eye catched this detail... MQTT Autodiscovery Setup _From 2023.1 **beta**_

So i guess, since i'm not on beta, this is never going to work with my stable version: 2023.1 and so one **needs to update!**

![image](https://github.com/zwave-js/zwave-js-ui/assets/25507805/e881c52a-42e8-4c4a-99a3-6ce21a8d477d)
